apollo-router/src/services/layers/persisted_queries/mod.rs

```diff
@@ -677,7 +660,7 @@ mod tests {
     async fn pq_layer_freeform_graphql_with_safelist() {
         let manifest = HashMap::from([(
             "valid-syntax".to_string(),
-            "fragment A on T { a }    query SomeOp { ...A ...B }    fragment,,, B on U{b c  } # yeah"
+            "fragment A on Query { me { id } }    query SomeOp { ...A ...B }    fragment,,, B on Query{me{name,username}  } # yeah"
```

Comment:
At some point while working on this PR I tried to make this code rely on `ExecutableDocument` instead of AST, which required queries to be valid for some schema. This turned out to cause other tests to fail because some checks happened earlier in the request pipelines, so I reverted back to AST. Quite possibly these test changes turned out unnecessary then.

After we fully migrate away from graphql-js validation and intentionally run apollo-rs validation early in the pipeline, I still plan to reduce AST usage and migrate things to `ExecutbleDocument` as much as possible.

Reply:
Makes sense. For the purpose of safelisting I think we should try to keep the safelist check before the validity check, though.